src/option_parser.cr

```diff
@@ -1,3 +1,30 @@
+# `OptionParser` is a class for command-line options processing. It supports:
+#
+# * Short and long modifier style options (example: `-h`, `--help`)
+# * Command style arguments (example: `crystal build`)
+# * Passing arguments to the flags (example: `-f filename.txt`)
+# * Automatic help message generation
+#
+# Run `crystal` for an example of a CLI built with `OptionParser`.
```

Comment:
is it? I believe it is not built with OptParser [source](https://github.com/manastech/crystal/blob/master/src/compiler/crystal/command.cr) - it's been on my "maybe change this and make a PR" list :)